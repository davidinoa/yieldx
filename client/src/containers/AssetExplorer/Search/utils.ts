interface InGridFilter {
  filter: string | number;
  filterTo: number;
  filterType: string;
  type: string;
  dateFrom: string;
  dateTo: string;
  values: string[];
}

export interface AgGridFilters {
  [key: string]: any;
}

const getFilter = (filter: InGridFilter) => {
  /**
   *  The filter object from Ag Grid API. In the case of filtering `name` column.
   *  { filter: "Asset Name", filterType: "text", type: "contains"}
   */
  let searchFilter = {};
  const {
    filter: filterContent,
    filterType,
    type,
    filterTo,
    values,
    dateFrom,
    dateTo
  } = filter;
  if (filterType === "text") {
    switch (type) {
      case "contains":
        searchFilter = { containsText: filterContent };
        break;
      case "notContains":
        searchFilter = { doesNotContainText: filterContent };
        break;
      case "equals":
        searchFilter = { equalsText: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualText: filterContent };
        break;
      case "startsWith":
        searchFilter = { startsWithText: filterContent };
        break;
      case "endsWith":
        searchFilter = { endsWith: filterContent };
        break;
      default:
        searchFilter = { containsText: filterContent };
        break;
    }
  } else if (filterType === "number") {
    switch (type) {
      case "equals":
        searchFilter = { equalsNumber: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualNumber: filterContent };
        break;
      case "lessThan":
        searchFilter = { isLessThanNumber: filterContent };
        break;
      case "lessThanOrEqual":
        searchFilter = { isLessThanOrEqualToNumber: filterContent };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanNumber: filterContent };
        break;
      case "greaterThanOrEqual":
        searchFilter = { isGreaterThanOrEqualToNumber: filterContent };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToNumber: filterContent,
          isLessThanOrEqualToNumber: filterTo
        };
        break;
      default:
        searchFilter = { equalsNumber: filterContent };
        break;
    }
  } else if (filterType === "date") {
    const convertedDateFrom = new Date(dateFrom);
    const convertedDateTo = new Date(dateTo);
    switch (type) {
      case "equals":
        searchFilter = { equalsDate: convertedDateFrom };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanDate: convertedDateFrom };
        break;
      case "lessThan":
        searchFilter = { isLessThanDate: convertedDateFrom };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualDate: convertedDateFrom };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToDate: convertedDateFrom,
          isLessThanOrEqualToDate: convertedDateTo
        };
        break;
      default:
        searchFilter = { equalsDate: convertedDateFrom };
        break;
    }
  } else if (filterType === "set" && values) {
    // We may need to add `set` type Ag grid column filter params like cusip or isin
    // in the search request in the future.
    // searchFilter = { inText: values };
  }
  return searchFilter;
};

export const convertInGridFilters = (filters: AgGridFilters) => {
  const convertedFilters: { [index: string]: any } = {};
  Object.keys(filters).forEach(key => {
    const inGridFilter = filters[key];
    if (key === "minimumQuantity") key = "bond.minimumDenomination";
    convertedFilters[key] = getFilter(inGridFilter);
  });

  return convertedFilters;
};
