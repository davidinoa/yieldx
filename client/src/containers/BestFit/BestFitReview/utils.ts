import { getRatingValue } from "../../../util/ratings";
import YieldXPortfolio, {
  YieldXPosition
} from "../../../models/YieldXPortfolio";

export function getBondRatingsData(
  portfolio: YieldXPortfolio
): { name: string; value: number }[] {
  if (portfolio.getBondSleeve) {
    const ratingMarketValues: { [key: string]: number } = {};
    const bondPortfolio = portfolio.getBondSleeve();
    bondPortfolio.positions.forEach((position: YieldXPosition) => {
      const { asset, weight } = position;
      const rating = asset.rating ?? "NR";
      ratingMarketValues[rating] = (ratingMarketValues[rating] || 0) + weight;
    });
    return Object.entries(ratingMarketValues)
      .map(([rating, weight]) => {
        return {
          name: rating,
          value: weight
        };
      })
      .sort((a, b) => {
        return getRatingValue(a.name) - getRatingValue(b.name);
      });
  }
  return [];
}
