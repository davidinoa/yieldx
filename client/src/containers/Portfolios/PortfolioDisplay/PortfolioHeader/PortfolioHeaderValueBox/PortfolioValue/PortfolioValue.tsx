import React from "react";
import { Row } from "react-bootstrap";
import { ValueTitle, ValueTotal } from "../../PortfolioHeader.styles";
import Decimal from "../../../../../../util/Decimal";

interface Props {
  value: number;
}

export default function PortfolioValue({ value }: Props) {
  const [dollars, cents] = Decimal.format(value).split(
    "."
  );
  return (
    <>
      <Row>
        <ValueTitle>
          <h3>PORTFOLIO VALUE</h3>
        </ValueTitle>
      </Row>
      <Row>
        <ValueTotal className="col">
              <span>
                {`$${dollars}`}
                <sup>{`.${cents}`}</sup>
              </span>
        </ValueTotal>
      </Row>
    </>
  );
}
