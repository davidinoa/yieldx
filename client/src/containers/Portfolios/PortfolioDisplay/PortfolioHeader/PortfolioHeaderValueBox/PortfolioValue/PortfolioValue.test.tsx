import React from "react";
import { render, screen } from "../../../../../../test/utils";
import PortfolioValue from "./PortfolioValue";

describe("PortfolioValue", () => {
  test("Should display title", () => {
    render(<PortfolioValue value={971.26} />);
    expect(screen.queryByText(/portfolio value/i)).toBeInTheDocument();
  });

  test("Should display correct dollar and cent values", () => {
    render(<PortfolioValue value={971.26} />);
    expect(screen.queryByText("$971")).toBeInTheDocument();
    expect(screen.queryByText(".26")).toBeInTheDocument();
  });

  test("Should match snapshot markup", () => {
    const { asFragment } = render(<PortfolioValue value={971.26} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
