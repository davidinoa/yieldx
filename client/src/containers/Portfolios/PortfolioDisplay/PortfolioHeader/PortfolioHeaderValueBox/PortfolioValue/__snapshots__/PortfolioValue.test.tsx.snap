// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioValue Should match snapshot markup 1`] = `
<DocumentFragment>
  .c0 {
  background: rgba(0,0,255,0.3);
  border-radius: 0.35rem 0.35rem 0 0;
  background: var(--dark);
  padding: 0.35rem;
}

.c0 h3 {
  color: var(--lightest);
  text-align: center;
  line-height: 1em;
  margin: 0;
  font-size: 0.75rem;
}

@media (min-width:1600px) {
  .c0 h3 {
    font-size: 0.75rem;
  }
}

@media (min-width:1600px) {

}

@media (min-width:1600px) {

}

<div
    class="row"
  >
    <div
      class="c0 col-24"
    >
      <h3>
        PORTFOLIO VALUE
      </h3>
    </div>
  </div>
  .c0 {
  position: relative;
  padding-top: 0.85rem;
}

.c0 span {
  display: block;
  text-align: center;
  font-size: 1.7rem;
  font-weight: bold;
  padding-bottom: 0.45rem;
}

.c0 span > sup {
  font-size: 0.55em;
  line-height: 22px;
  padding: 0.25rem;
}

@media (min-width:1600px) {

}

@media (min-width:1600px) {
  .c0 {
    padding-top: 1.25rem;
  }
}

@media (min-width:1600px) {
  .c0 span {
    font-size: 2rem;
  }
}

<div
    class="row"
  >
    <div
      class="c0 col"
    >
      <span>
        $971
        <sup>
          .26
        </sup>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
